internal/login/login.go

```diff
@@ -0,0 +1,106 @@
+package login
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/launchdarkly/ldcli/internal/errors"
+)
+
+const ClientID = "e6506150369268abae3ed46152687201"
+
+type DeviceAuthorization struct {
+	DeviceCode      string `json:"deviceCode"`
+	ExpiresIn       int    `json:"expiresIn"`
+	UserCode        string `json:"userCode"`
+	VerificationURI string `json:"verificationUri"`
+}
+
+type UnauthenticatedClient interface {
+	MakeRequest(
+		method string,
+		path string,
+		data []byte,
+	) ([]byte, error)
+}
+
+type Client struct {
+	cliVersion string
+}
+
+func NewClient(cliVersion string) Client {
+	return Client{
+		cliVersion: cliVersion,
+	}
+}
+
+func (c Client) MakeRequest(
+	method string,
+	path string,
+	data []byte,
+) ([]byte, error) {
+	client := http.Client{}
+
+	req, _ := http.NewRequest(method, path, bytes.NewReader(data))
+	req.Header.Add("Content-type", "application/json")
+	req.Header.Set("User-Agent", fmt.Sprintf("launchdarkly-cli/v%s", c.cliVersion))
+
+	res, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
```

Comment:
this should be called as soon as we know `res.Body` is not nil in case there is an error with reading, right?

Reply:
I believe the idea is that res.Body will be nil if there's an error, so there's nothing to close. There's a [linter](https://github.com/timakin/bodyclose) with an example that closes the body after the error check.